integration/tests/services/test_ws_subscriber.py

```diff
@@ -0,0 +1,223 @@
+                arg_topics = []
+                for item in arg_filter:
+                    arg_topics.append("0x" + keccak(text=item).hex())
+                topics.append(arg_topics)
+
+            optional_fields = {"address": event_caller.address}
+            if topics:
+                optional_fields["topics"] = topics
+            params = ["logs", optional_fields]
+
+            data = Subscribe(params)
+            await ws.send(json.dumps(data.__dict__))
+            r = await ws.recv()
+            response = json.loads(r, object_hook=lambda d: SimpleNamespace(**d))
+            subscription = response.result
+            assert len(subscription) > 0
+
+            tx = self.make_tx_object(self.sender_account.address)
+            instruction_tx = event_caller.functions.callEvent1(arg1).build_transaction(tx)
+            self.web3_client.send_transaction(self.sender_account, instruction_tx)
+
+            tx = self.make_tx_object(self.sender_account.address)
+            instruction_tx = event_caller.functions.callEvent2(arg1, arg2).build_transaction(
+                tx
+            )
+            self.web3_client.send_transaction(self.sender_account, instruction_tx)
+
+            tx = self.make_tx_object(self.sender_account.address)
+            instruction_tx = event_caller.functions.callEvent2(arg2, arg3).build_transaction(
+                tx
+            )
+            self.web3_client.send_transaction(self.sender_account, instruction_tx)
+
+            tx = self.make_tx_object(self.sender_account.address)
+            instruction_tx = event_caller.functions.callEvent3(
+                arg1, arg2, arg3
+            ).build_transaction(tx)
+            self.web3_client.send_transaction(self.sender_account, instruction_tx)
+
+            messages = await ws_receive_all_messages(ws)
+            assert (len(messages) == log_count), f"Expected {log_count} event logs, but found {len(messages)}"
+
+            is_event_topic_in_list = False
+            is_arg_topic_in_list = False
+            for m in messages:
+                log = m.params.result
+                if topics[0]:
+                    for topic in topics[0]:
+                        if topic in log.topics:
+                            is_event_topic_in_list = True
+                else:
+                    is_event_topic_in_list = True
+                if len(topics) == 2:
+                    for topic in topics[1]:
+                        if topic in log.topics:
+                            is_arg_topic_in_list = True
+                else:
+                    is_arg_topic_in_list = True
+
+            assert (is_event_topic_in_list), f"Filter by {topics} works incorrect. Response: {m}"
+            assert (is_arg_topic_in_list), f"Filter by {topics} works incorrect. Response: {m}"
+
+    @pytest.mark.parametrize("subscription_type", ["newHeads", "logs"])
+    async def test_another_user_cant_unsubscribe(self, subscription_type: string):
```

Comment:
we need a test when a several users connect to one contract and subscribe to several event types and get only this

Reply:
Added